hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperation.java

```diff
@@ -463,8 +268,182 @@ protected void writeInternal(ObjectDataOutput out) throws IOException {
         out.writeObject(entryProcessor);
     }
 
-    @Override
-    public int getId() {
-        return MapDataSerializerHook.ENTRY_OPERATION;
+    private final class EntryOperationOffload extends Offload {
+        private EntryOperationOffload() {
+            super(EntryOperation.this);
+        }
+
+        @Override
+        public void start() {
+            verifyEntryProcessor();
+
+            boolean shouldCloneForOffloading = OBJECT.equals(mapContainer.getMapConfig().getInMemoryFormat());
+            Object oldValue = recordStore.get(dataKey, false);
+            Object clonedOldValue = shouldCloneForOffloading ? serializationService.toData(oldValue) : oldValue;
+
+            String executorName = ((Offloadable) entryProcessor).getExecutorName();
+            executorName = executorName.equals(Offloadable.OFFLOADABLE_EXECUTOR) ? OFFLOADABLE_EXECUTOR : executorName;
+
+            if (readOnly) {
+                executeReadOnlyEntryProcessor(clonedOldValue, executorName);
+            } else {
+                executeMutatingEntryProcessor(clonedOldValue, executorName);
+            }
+        }
+
+        private void verifyEntryProcessor() {
```

Comment:
Yes we do. The start method is executed within the main try/catch block of operation execution and failure handling is the same as with an exception thrown by operation.run. 

Also when the Offloaded.init is called, the existing response handler is decorated with the OffloadedOperationResponseHandler which in case of failure will remove the operation from the asynchronous operations list.